src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SemanticFacts/CSharpSemanticFacts.cs

```diff
@@ -237,7 +237,7 @@ private static void FlattenDeconstructionMethods(DeconstructionInfo deconstructi
         public bool IsPartial(ITypeSymbol typeSymbol, CancellationToken cancellationToken)
         {
             var syntaxRefs = typeSymbol.DeclaringSyntaxReferences;
-            return syntaxRefs.Any(static (n, cancellationToken) => ((BaseTypeDeclarationSyntax)n.GetSyntax(cancellationToken)).Modifiers.Any(SyntaxKind.PartialKeyword), cancellationToken);
+            return syntaxRefs.Any(static (n, cancellationToken) => (n.GetSyntax(cancellationToken) as BaseTypeDeclarationSyntax)?.Modifiers.Any(SyntaxKind.PartialKeyword) == true, cancellationToken);
```

Comment:
`typeSymbol.DeclaringSyntaxReferences` is doc'd as possibly returning CompilationUnitSyntax so it seemed like the unconditional cast was not safe in general

Reply:
that is true.  if it's not a big deal, can you also make it typed to be INamedTypeSymbol?

Reply:
i'd doc that we may not always have a type decl here.

Reply:
oh, this is bad on multiple levels.  anotehr case is delegates, which this would throw on.  

finally, consdier just rewriting as a foreach loop.  i think it would be cleaner.